iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -1085,20 +1241,25 @@ protected override void Dispose(bool disposing)
 
                 if (!_isDisposed)
                 {
+                    _isDisposing = true;
+
                     base.Dispose(disposing);
                     if (disposing)
                     {
                         _handleDisconnectCts?.Cancel();
```

Comment:
We have 2 places where we cancel, which means one of them will get an exception, which we should gracefully handle.

Reply:
Looks like `CancellationTokenSource` gracefully handles multiple calls to the `Cancel()` method. From the [sourcecode](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs):

```c#
public void Cancel() => Cancel(false);

public void Cancel(bool throwOnFirstException)
{
	ThrowIfDisposed();
	NotifyCancellation(throwOnFirstException);
}

private void NotifyCancellation(bool throwOnFirstException)
{
	if (TransitionToCancellationRequested())
	{
		// If we're the first to signal cancellation, do the main extra work.
		ExecuteCallbackHandlers(throwOnFirstException);
		Debug.Assert(IsCancellationCompleted, "Expected cancellation to have finished");
	}
}

private bool TransitionToCancellationRequested()
{
	if (!IsCancellationRequested &&
		Interlocked.CompareExchange(ref _state, NotifyingState, NotCanceledState) == NotCanceledState)
	{
		// Dispose of the timer, if any.  Dispose may be running concurrently here, but TimerQueueTimer.Close is thread-safe.
		TimerQueueTimer? timer = _timer;
		if (timer != null)
		{
			_timer = null;
			timer.Close();
		}

		// Set the event if it's been lazily initialized and hasn't yet been disposed of.  Dispose may
		// be running concurrently, in which case either it'll have set m_kernelEvent back to null and
		// we won't see it here, or it'll see that we've transitioned to NOTIFYING and will skip disposing it,
		// leaving cleanup to finalization.
		_kernelEvent?.Set();

		return true;
	}

	return false;
}


```

The [first check](https://github.com/dotnet/runtime/blob/0a15c3b97dabaf94e5d4399aef2fd48c2b3f446a/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs#L676) in `TransitionToCancellationRequested` returns `false`, which will result in each loop being gracefully exited immediately.